dbt_semantic_interfaces/parsing/where_filter/parameter_set_factory.py

```diff
@@ -37,26 +44,46 @@ def create_time_dimension(
         entity_path: Sequence[str] = (),
         date_part_name: Optional[str] = None,
     ) -> TimeDimensionCallParameterSet:
-        """Gets called by Jinja when rendering {{ TimeDimension(...) }}."""
+        """Gets called by Jinja when rendering {{ TimeDimension(...) }}.
+
+        There is a lot of strangeness around the time granularity specification here. Historically,
+        we accepted time dimension names of the form `metric_time__week` or `customer__registration_date__month`
+        in this interface. We have not yet fully deprecated this, and it's unclear if we ever will.
+
+        Key points to note:
+          1. The time dimension name parsing only accepts standard time granularities. This will not change.
+          2. The time granularity parameter is what we want everybody to use because it's more explicit.
+          3. The time granularity parameter will support custom granularities, so that's nice
+
+        While this all may seem pretty bad it's not as terrible as all that - this class is only used
+        for parsing where filters. When we solve the problems with our current where filter spec this will
+        persist as a backwards compatibility model, but nothing more.
+        """
         group_by_item_name = DunderedNameFormatter.parse_name(time_dimension_name)
         if len(group_by_item_name.entity_links) != 1 and not is_metric_time_name(group_by_item_name.element_name):
             raise ParseWhereFilterException(
                 ParameterSetFactory._exception_message_for_incorrect_format(time_dimension_name)
             )
-        grain_parsed_from_name = group_by_item_name.time_granularity
-        grain_from_param = TimeGranularity(time_granularity_name) if time_granularity_name else None
+        grain_parsed_from_name = (
+            group_by_item_name.time_granularity.value if group_by_item_name.time_granularity else None
+        )
+        grain_from_param = time_granularity_name
```

Comment:
Is this line necessary? Why not just use `time_granularity_name` instead of using a different variable with the same value?